src/main/java/core/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.dao;
+
+import core.model.User;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserDaoImpl implements UserDao {
+    @Autowired
+    private SessionFactory sessionFactory;
```

Comment:
https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_01.html
> What is better: setter, constructor, or field injection? Think why.

Reply:
Changed to Constructor injection